core/unit_test/TestRange.hpp

```diff
@@ -332,19 +332,49 @@ TEST(TEST_CATEGORY, range_dynamic_policy) {
 #if !defined(KOKKOS_ENABLE_CUDA) && !defined(KOKKOS_ENABLE_HIP) && \
     !defined(KOKKOS_ENABLE_SYCL)
   {
-    TestRange<TEST_EXECSPACE, Kokkos::Schedule<Kokkos::Dynamic> > f(0);
+    TestRange<TEST_EXECSPACE, Kokkos::Schedule<Kokkos::Dynamic>> f(0);
     f.test_dynamic_policy();
   }
   {
-    TestRange<TEST_EXECSPACE, Kokkos::Schedule<Kokkos::Dynamic> > f(3);
+    TestRange<TEST_EXECSPACE, Kokkos::Schedule<Kokkos::Dynamic>> f(3);
     f.test_dynamic_policy();
   }
   {
-    TestRange<TEST_EXECSPACE, Kokkos::Schedule<Kokkos::Dynamic> > f(1001);
+    TestRange<TEST_EXECSPACE, Kokkos::Schedule<Kokkos::Dynamic>> f(1001);
     f.test_dynamic_policy();
   }
 #endif
 }
 #endif
 
+void test_large_parallel_for_reduce() {
+  using ExecutionSpace              = typename TEST_EXECSPACE::execution_space;
+  constexpr long long unsigned size = 1llu << 32;
+  Kokkos::View<long long unsigned, TEST_EXECSPACE::memory_space> v("v");
+
+  // We want to explicitly test that using a parallel_for for filling the View
+  // works
+  Kokkos::parallel_for(
+      Kokkos::RangePolicy<ExecutionSpace,
+                          Kokkos::IndexType<long long unsigned>>(0, size),
+      KOKKOS_LAMBDA(long long unsigned) { Kokkos::atomic_add(v.data(), 1); });
```

Comment:
This makes the test time out.

Reply:
I mean we are fixing `parallel_for` explicitly here.

Reply:
sry, my bad.
Hmm since we discovered this in `hip.view_allocation_large_rank` ... we could think about squeezing a test in there reusing the view the test allocates

Reply:
I just reverted the last commit so we at least have soemthing working. Feel free to push something better.